src/app/Dashboard/Diagnostics/DiagnosticsCard.tsx

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright The Cryostat Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  DashboardCardTypeProps,
+  DashboardCardFC,
+  DashboardCardSizes,
+  DashboardCardDescriptor,
+} from '@app/Dashboard/types';
+import { ErrorView } from '@app/ErrorView/ErrorView';
+import { FeatureLevel } from '@app/Shared/Services/service.types';
+import { LoadingProps } from '@app/Shared/Components/types';
+import { ServiceContext } from '@app/Shared/Services/Services';
+import { useSubscriptions } from '@app/utils/hooks/useSubscriptions';
+import {
+  Bullseye,
+  Button,
+  CardBody,
+  CardHeader,
+  CardTitle,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateIcon,
+  EmptyStateVariant,
+  Label,
+  EmptyStateHeader,
+  EmptyStateFooter,
+} from '@patternfly/react-core';
+import { DataSourceIcon, SyncAltIcon, TachometerAltIcon } from '@patternfly/react-icons';
+import * as React from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+import { DashboardCard } from '../DashboardCard';
+
+export interface DiagnosticsCardProps extends DashboardCardTypeProps {
+  chartKind: string;
+  duration: number;
+  period: number;
+}
+
+export enum DiagnosticsCardKind {}
+
+export function kindToId(kind: string): number {
+  return DiagnosticsCardKind[kind];
+}
+
+export const DiagnosticsCard: DashboardCardFC<DiagnosticsCardProps> = (props) => {
+  const { t } = useTranslation();
+  const serviceContext = React.useContext(ServiceContext);
+  const addSubscription = useSubscriptions();
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const [running, setRunning] = React.useState(false);
+  const isError = React.useMemo(() => errorMessage != '', [errorMessage]);
+
+  const handleError = React.useCallback(
+    (error) => {
+      setErrorMessage(error.message);
+    },
+    [setErrorMessage],
+  );
+
+  const handleGC = React.useCallback(() => {
+    addSubscription(
+      serviceContext.api.runGC().subscribe({
+        error: (err) => handleError(err),
+      }),
+    );
+  }, [addSubscription, serviceContext.api, handleError]);
+
+  const GCButton = React.useMemo(() => {
+    return (
+      <Button
+        key={0}
+        aria-label={t('DaignosticsCard.DIAGNOSTICS_GC_BUTTON', { chartKind: props.chartKind })}
+        onClick={handleGC}
```

Comment:
If I'm following correctly,

![image](https://github.com/user-attachments/assets/7f044eb3-4741-43ae-98fb-9a1b602e6fbf)

This `<Button>` with the `<SyncAltIcon />` is the "refresh" button appearing in the card header, and when clicked it performs the same action as the big clue "Start Garbage Collection" button?

I think for now we should just pick one button or the other, rather than both. I don't know for sure what's best to go with since it probably depends on what other diagnostic actions or data retrieval we might implement in the future, and whether or not those additions in the future are part of this same card or are their own separate cards. Anyway, we're flagging this as a Beta component, so I don't mind if the UI is not totally settled and ends up changing significantly in the future, so let's just pick one button or the other and stick with it.

If we stick with this "refresh" header button then I think it will at least need a hover tooltip or something to inform the user about what the button does.

Reply:
I think sticking with the GC button makes more sense, I'll remove the refresh button